### DISCUSSION TITLE
Some Folders are viewable, some aren't (m3U causes the issue)

### DISCUSSION THREAD

**Discussion by subatomicglue** [rern/rAudio#688] [Bug Reports]
**Any idea why some folders can be viewed, and some cannot?**

I have a USB stick with folders of mp3 and m4a files.
I've navigated to Library / USB / BELLA  (name of my USB stick)

Raspberry Pi 4b running 20211112

Shown in the video are the following folders, most wont open  (only 2 open to show the files inside):

- `Ambient Theology (Virus)`  (has m4a files inside)
- `aphex twin - 1992 - SelectedAmbientWorks 85-92`   (has mp3 files inside)
- `Daft Punk - Tron Legacy (Deluxe Edition)`  (has mp3 files, and opens to show the file listing when tapped on)
- `Daft Punk - Tron Legacy Reconfigured`    (has mp3 files, and opens to show the file listing when tapped on)
- `Mystical sun - Primordial Atmospheres`   (has mp3 files)
- `Tropical Rain Forest - Nature Sounds Music` (has a single m4a file)
- `VA - Excursions In Ambience - The Third Dimension (1994)`   (has mp3 files)

Sorry for the autofocus, hopefully you can see that most folders are unresponsive.   I would have to tap on the folder icon to Add or Replace the queue - but there is no way to view what's inside the folder  (except for 2 of the folders!)

**UPDATE**: we found that m3U file present in the folder causes the issue

https://user-images.githubusercontent.com/7660853/144725065-f41e44b9-6626-4bcc-8fc0-dac19b14beb0.MOV

**Comment by sp300b** [comment]
Hi, 

Have you checked this one https://github.com/rern/rAudio-1/discussions/270#discussioncomment-540788

**Reply by subatomicglue** [reply]
Interesting, for sure, I have an m3U in every folder, I actually wrote a Perl script to generate them recursively, for my entire collection.   

Odd that rAudio-1 would get tripped up by this!  Seems like an undesirable feature (bug...).  

I also noticed that I can't tap on an m3U file, rAudio-1 doesn't know what to do with it, so I may as well delete them all....    odd that it wouldn't just queue them

**Reply by subatomicglue** [reply]
Status: fixed (for me).  Sure enough, deleting all my m3U files fixes the folder view issues I reported.   

Of course, deleting all the .m3U files wouldn't work for my main music collection on my NAS, since I need those files for several other apps I serve to...

Fortunately on the USB stick, I can delete them all.


Still a bug that rAudio-1 should fix:
---> avoiding weird folder behavior when m3U file is present.

**Reply by sp300b** [reply]
I had yesterday basically the same problem as been reported above on the top  - not opening some music folders, which was at the end connected to having unnecessary .CUE files in a such a folders. Inside the folders where .CUE files were really necessary, system responded as expected, opening the folders and playing the music.
I actually found explanations here (https://github.com/rern/rAudio-1/discussions/591#discussioncomment-1631790)
Could be that the same story goes for .M3U files.
Regards, sp300b

**Comment by ceejayemm** [comment]
I can't view the movie for some reason and I am not clear if it is only the folders which contain the mp3 files which are visible / playable. If it is the m4a files which are causing you problems do you have the (Player) FFmpeg decoder option turned on ?

(Sorry if this is too obvious)

Chris

**Reply by subatomicglue** [reply]
I think GitHub doesn't preview the movie, but you can download it and then view

**Reply by subatomicglue** [reply]
My iPhone plays the movie inline, actually......

**Reply by subatomicglue** [reply]
The other post reply here suggested to remove all my m3U playlist files in those folders

Looks like rAudio-1 has a bug with m3U files that screws up the folder view capabilities.  

An m3U file present in the folder makes it so you can't view what's in that folder (nor display the playlist either, so I don't understand the point - thus, it seems to be a bug).

**Reply by phreatic62** [reply]
Thanks for posting this, deleting .m3u files works for me too. My problem was that I could see some albums in 'Album' view, but would not get any response if I clicked on them.

**Reply by rern** [reply]
**rAudio** parses `*.m3u` files with `mpc -f %FORMAT% playlist "MPDPATH/file.m3u"`:

![m3u](https://user-images.githubusercontent.com/10094236/144847550-f7fb468c-5f72-4e5b-8dd9-91579b593655.jpg)

Please explain advantages of M3U over ID3 tags and how to create ones with all metadata.

**Reply by subatomicglue** [reply]
I replied at the top level instead of here... since It's a long reply :)   The tldr; is that m3U shouldn't do anything at all unless tapped on, and definitely shouldn't create an untappable folder that does nothing...

**Comment by subatomicglue** [comment]
```
rAudio parses *.m3u files with mpc -f %FORMAT% playlist "MPDPATH/file.m3u":
Please explain advantages of M3U over ID3 tags and how to create ones with all metadata.
```
rern,

In my folder structure:
ID3 tags are kept in `mp3` and `m4a` files.   NOT in the `.m3U` playlist files.  
My use of `m3U` playlist files is as pure playlist only (no tagging needs here).

Keeping in mind that my NAS is used between **rAudio** as well as other computers on my network, **plex**, **VLC**, **winamp**, etc...
- And that my USB drive is a copy of my NAS folders...
- And that I'd still like to connect to my NAS with **rAudio-1**, feels like a capability that should work....

I have `m3U` files for every folder in my NAS's `music/` partition, each one is called `!all.m3U`, and represents all mp3/m4a files below that directory.     I generate the `m3U` files with a **perl** script that recurses through the directories.    

For example

- If I wanted to play all albums in the folder "`Tool (albums)/`", then I would click on "`Tool (albums)/!all.m3U`"
  - which would open in **VLC** player on **MacOS**, or **winamp** in **Windows**...
- If I wanted to play all music on the NAS, I could click on the top level folder's `music/!all.m3U`...
- If I wanted to play just one album, I could click on "`Tool (albums)/1993 Undertow/!all.m3U`"

On the **rAudio-1**, I would want these `m3U` files to be ignored unless I tapped on them, and then I would hope for the same functionality as **winamp**/**vlc** has when clicking on the `m3U` file: which is to queue the `m3U`'s listing into the **rAudio1**'s playlist queue..    

- I would _not_ expect that I couldn't descend into the folder at all, no listing... (for folders with m3U files)
  - ...this is the problem I was seeing in the OP above...
- I would _not_ expect that descending into the folder would show the `m3U` listing
- I would _not_ expect the `m3U` to somehow change or augment the mp3/m4a file tags in the folder
- If these are features - then I would hope these would be feature options I can change/customize to make my **rAudio1** box more "sane" for me and my family to use :)

As an aside:   I tend only to use folder view, and I never use artist or album view.   It's just kind of chaos since tags aren't always consistent (both my fault from 25 years ago and the dawn of 'ripping' CDs back then, as well as iTunes/Amazon/CDDB's fault where some albums dont have consistent naming and especially for "various artists" albums, etc...).   So I tend to stick to folder views.   If I could I would disable Artist/Album views... :)   Or...  if I had something like Plex does to fetch and fix information, I'd use that (even Plex isn't perfect - and I use folders with Plex too).

Example `!all.m3U` file for the location `Tool (albums)/1993 Undertow/!all.m3U`:
```
#EXTM3U
01 Intolerance.m4a
02 Prison Sex.m4a
03 Sober.m4a
04 Bottom.m4a
05 Crawl Away.m4a
06 Swamp Song.m4a
07 Undertow.m4a
08 4 Degrees.m4a
09 Flood.m4a
69 Disgustipated.m4a
```

**Reply by tjengbudi** [reply]
hello, do you have any solution of this problem?
or this just ignored?
